app/views/portal/show.rb

```diff
@@ -0,0 +1,871 @@
+                  begin
+                    date = Time.parse(extra_val)
+                    formatted = date.strftime(xtra[:format_date])
+                    extra_val = formatted
+                  rescue
+                  end
+                end
+                extra_info_builder = extra_info
+                path_segments = xtra[:map_to] || xtra[:field]
+                path_segments = path_segments.split('.')
+
+                path_segments.each.map do |path_segment|
+                  is_last = path_segment == path_segments.last
+                  extra_info_builder[path_segment] ||= (is_last ? extra_val : {})
+                  extra_info_builder = extra_info_builder[path_segment]
+                end
+                ob['extra_info'] = extra_info
+              end
+            end
+
+            ob['text'] = text
+            subsection << ob unless text.nil? || text.blank?
+          end
+
+          if subsection.size > 0
+            section_data << subsection
+            section_labels << (section[:title].nil? ? false : t(section[:title]))
+          end
+        end
+      end
+
+      {
+        title: t(data[:title]),
+        sections: section_data.each_with_index.map do |subsection, index|
+          if subsection.size > 0
+            {
+              title: section_labels[index],
+              items: subsection
+            }
+          else
+            false
+          end
+        end
+      } unless section_data.size == 0
+    end
+
+    # def content_object_download
+    #   links = []
+
+    #   if edm_is_shown_by_download_url.present?
+    #     links << {
+    #       text: t('site.object.actions.download'),
+    #       url: edm_is_shown_by_download_url
+    #     }
+    #   end
+
+    #   return nil unless links.present?
+
+    #   {
+    #     primary: links.first,
+    #     secondary: {
+    #       items: (links.size == 1) ? nil : links[1..-1]
+    #     }
+    #   }
+    # end
+
+    def long_and_lat?
+      latitude = render_document_show_field_value(document, 'places.latitude')
+      longitude = render_document_show_field_value(document, 'places.longitude')
+      !latitude.nil? && latitude.size > 0 && !longitude.nil? && longitude.size > 0
+    end
+
+    def session_tracking_path_opts(counter)
+      {
+        per_page: params.fetch(:per_page, search_session['per_page']),
+        counter: counter,
+        search_id: current_search_session.try(:id)
+      }
+    end
+
+    def doc_title
+      # force array return with empty default
+      title = document.fetch(:title, nil)
+
+      if title.blank?
+        render_document_show_field_value(document, 'proxies.dcTitle')
+      else
+        title.first
+      end
+    end
+
+    # Media type function normalises mime types
+    # Current @mime_type variable only workd for edm_is_shown_by
+    # Once it works for web_resources we change this function so
+    # that  it accepts a mime type rather than a url, and
+
+    def media_type(url)
```

Comment:
Assignment Branch Condition size for media_type is too high. [15.94/15]<br>Method has too many lines. [14/10]